### DISCUSSION TITLE
Use xvector as segmentation model's input?

### DISCUSSION THREAD

**Discussion by xIaott-s** [pyannote/pyannote-audio#885] [Ideas]
I was wondering is that possible to use xvector instead of raw audio as the segmentation model's input? 
This may make the model more robust?

**Comment by hbredin** [comment]
This would need a bit of work on your side but that is possible, yes.

1. [Create a new model](https://github.com/pyannote/pyannote-audio/blob/develop/tutorials/add_your_own_model.ipynb) architecture in lieu of [PyanNet](https://github.com/pyannote/pyannote-audio/blob/79fe1eafa98020e354bacd09e1b62cb820446364/pyannote/audio/models/segmentation/PyanNet.py#L38) that, for instance, replaces its [SincNet](https://github.com/pyannote/pyannote-audio/blob/79fe1eafa98020e354bacd09e1b62cb820446364/pyannote/audio/models/segmentation/PyanNet.py#L93) layer by a pretrained XVector extractor
2. [Train](https://github.com/pyannote/pyannote-audio/blob/develop/tutorials/training_a_model.ipynb) this new model
3. Profit!

**Reply by xIaott-s** [reply]
Thanks，I will try and let you know the result.

**Comment by xIaott-s** [comment]
I use a frame level xvetor embedding instead of SincNet. In my speaker change detection testset, it works better.
But I have a question that why do you choose the training sample from audios randomly in every epoch?

**Reply by hbredin** [reply]
If possible, would you mind sharing the code of this new architecture? It could be useful for other segmentation tasks as well.

Regarding random sampling, what is the alternative? Using a sliding window over the whole training set? That would always generate the same exact chunks (start and end times) over and over again. 

The idea of random sampling is to maximize variability of training samples -- you may see this as a kind of (weak) data augmentation.
